api/src/main/java/com/spotify/ffwd/Plugin.java

```diff
@@ -0,0 +1,27 @@
+package com.spotify.ffwd;
+
+import com.spotify.ffwd.filter.Filter;
+import com.spotify.ffwd.filter.TrueFilter;
+import java.util.Optional;
+import lombok.AccessLevel;
+import lombok.Data;
+import lombok.RequiredArgsConstructor;
+
+@Data
+public abstract class Plugin {
+    protected final String id;
+    protected final Optional<Filter> filter;
+
+    protected Plugin(final String id, final Optional<Filter> filter) {
+        this.id = id;
+        this.filter = filter.isPresent() ? filter : Optional.of(new TrueFilter());
```

Comment:
`filter` field should be empty if not specified to prevent the delegator being set up in `OutputPlugin`.